hatchways/team-bamboo
# Socket.io Setup

### Closes #30 

### What this PR does (required):

### Client
#### Minor updates
##### 1. useSocketContext -> now relies on auth provider and manages the connection to a socket, depending on whether or not the user is signed in.
- disconnectSocket -> a helper method used to forcefully disconnect the user from the socket.
---
### Server
#### Minor updates
##### 1. protectSocket -> specific middleware function which ensures a user is signed in before being able to connect to the socket.
##### 2. Added socket.io-cookie-parser middleware to parse socket.io's cookies.

### Any information needed to test this feature (required):
- Ensure that both the client and server are running.
- visit website (doesn't matter which page).
- check server logs, if you're already signed in, should log the user's name and the socket's id.
- if not logged in, nothing should appear in server console
- if you log out through website, socket will be disconnected and logged server side.

### Additional Notes
##### 1. I chose to make the socket context handle whether or not the socket connection is established. To ensure, that no matter what page a signed in user visits, the socket will be established, meaning the user should expect immediate interactivity, without needing to visit the the user's dashboard.
